### PR TITLE
refactor(misc): pretty print namespace.iD and peer.ID

### PR DESCRIPTION
**Pull request by walldiss**
## Overview

Log `namespace.ID` and `peer.ID` in pretty format. Also opened an issue for nmt pkg to add `StringHex` method and if it is **OK**, will update PR to use it. 

https://github.com/celestiaorg/nmt/issues/200

### CODE REVIEW

**Review by distractedm1nd — APPROVED** [review]
Thanks for this

File: share/getters/shrex.go [hunk 1]
```diff
@@ -204,6 +205,7 @@ func (sg *ShrexGetter) GetSharesByNamespace(
 		if getErr != nil {
 			log.Debugw("nd: couldn't find peer",
 				"hash", root.String(),
+				"nid", hex.EncodeToString(id),
```
Comment (renaynay) [inline share/getters/shrex.go:208]: we should eventually have method on namespace.ID to pretty print the hex right?

File: share/p2p/peers/manager.go [hunk 2]
```diff
@@ -390,15 +390,14 @@ func (m *Manager) getOrCreatePool(datahash string) *syncPool {
...
 	m.metrics.observeBlacklistPeers(reason, len(peerIDs))
 
-	if !m.params.EnableBlackListing {
-		return
-	}
 	for _, peerID := range peerIDs {
+		log.Debugw("blacklisting peer", "peer", peerID.String(), "reason", reason)
+		if !m.params.EnableBlackListing {
+			continue
+		}
```
Comment (renaynay) [inline share/p2p/peers/manager.go:399]: can you put debug log after this then?

**Review by renaynay — APPROVED** [review]

File: share/getters/shrex.go (hunk 1 above)
Comment (walldiss) [reply share/getters/shrex.go:208]: It is already merged and will be there in next nmt release.

File: share/p2p/peers/manager.go (hunk 2 above)
Comment (walldiss) [reply share/p2p/peers/manager.go:399]: I made it before, to make it possible to debug blacklisting while it is disabled.

Comment (renaynay) [reply share/p2p/peers/manager.go:399]: can you make a note of that then 😬  hopefully we'll remove this EnableBlacklisting altogether soon

Comment (walldiss) [reply share/p2p/peers/manager.go:401]: I've added log line, to clarify why it is always logged

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**